src/controller/CHIPDeviceController.cpp

```diff
@@ -207,13 +207,11 @@ CHIP_ERROR DeviceController::Init(NodeId localDeviceId, ControllerInitParams par
     mExchangeMgr->SetDelegate(this);
 
 #if CHIP_DEVICE_CONFIG_ENABLE_MDNS
-    if (params.mDeviceAddressUpdateDelegate != nullptr)
```

Comment:
Why is this null-check being removed?

Reply:
Because all calls later on to mDeviceAddressUpdateDelegate are already conditional on whether it is null or not, so an additional null check isn't needed.